ElectronicObserverTypes/Attacks/Extensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace ElectronicObserverTypes.Attacks;
+
+public static class Extensions
+{
+	public static bool IsSpecialAttack(this DayAttackKind dayAttack) => dayAttack is
+		DayAttackKind.SpecialNelson or
+		DayAttackKind.SpecialNagato or
+		DayAttackKind.SpecialMutsu or
+		DayAttackKind.SpecialColorado or
+		DayAttackKind.SpecialKongo or
+		DayAttackKind.SpecialSubmarineTender23 or
+		DayAttackKind.SpecialSubmarineTender34 or
+		DayAttackKind.SpecialSubmarineTender24 or
+		DayAttackKind.SpecialYamato2Ships or
+		DayAttackKind.SpecialYamato3Ships;
+
+	public static bool IsSpecialAttack(this NightAttackKind nightAttack) => nightAttack is
+		NightAttackKind.CutinZuiun or
+		NightAttackKind.SpecialNelson or
+		NightAttackKind.SpecialNagato or
+		NightAttackKind.SpecialMutsu or
+		NightAttackKind.SpecialColorado or
+		NightAttackKind.SpecialKongou or
+		NightAttackKind.SpecialSubmarineTender23 or
+		NightAttackKind.SpecialSubmarineTender34 or
+		NightAttackKind.SpecialSubmarineTender24 or
+		NightAttackKind.SpecialYamato2Ships or
+		NightAttackKind.SpecialYamato3Ships;
+
+	public static List<int> SpecialAttackParticipationIndexes(this DayAttackKind? dayAttack)
+		=> dayAttack switch
+		{
+			{ } atk => atk switch
+			{
+				DayAttackKind.SpecialSubmarineTender23 or
+				DayAttackKind.SpecialSubmarineTender34 or
+				DayAttackKind.SpecialSubmarineTender24 => SpecialAttackIndexes(atk)
+					.Append(0)
+					.Distinct()
```

Comment:
This code doesn't really make sense to me. The idea of the special attack indexes function was to just provide the exact indexes. If the sub code needs special handling, that should happen in the battle detail code.

Reply:
Attack participants and attackers are different things, sub tenders are attack participants even though they don't attack in their special.